### PR TITLE
feat: implement use the '-' mark to ignore writing values for field

### PR DESCRIPTION
**Pull request by daviderli614**
## What's changed and what's your intention?
Similar to the following struct, The CPU field is marked with `-`,  will write the zero value of the type:
```
type Monitor struct {
	ID          int64     `greptime:"tag;column:id;type:int64"`
	Host        string    `greptime:"tag;column:host;type:string"`
	Memory      uint64    `greptime:"field;column:memory;type:uint64"`
	Cpu         float64   `greptime:"-"`
	Temperature int64     `greptime:"field;column:temperature;type:int64"`
	Running     bool      `greptime:"field;column:running;type:boolean"`
	Ts          time.Time `greptime:"timestamp;column:ts;type:timestamp;precision:millisecond"`
}
```
Result:
```
+------+-----------+--------+------+-------------+---------+----------------------------+
| id   | host      | memory | cpu  | temperature | running | ts                         |
+------+-----------+--------+------+-------------+---------+----------------------------+
|    0 | 127.0.0.1 |      1 |    0 |          -1 |       0 | 2024-12-02 07:31:46.403000 |
|    0 | 127.0.0.1 |      1 |    0 |          -1 |       0 | 2024-12-02 07:31:46.587000 |
|    1 | 127.0.0.2 |      1 |    0 |          -1 |       1 | 2024-12-02 07:31:46.403000 |
|    1 | 127.0.0.2 |      1 |    0 |          -1 |       1 | 2024-12-02 07:31:46.587000 |
|    2 | 127.0.0.3 |      2 |    0 |          -2 |       1 | 2024-12-02 07:31:46.403000 |
|    2 | 127.0.0.3 |      2 |    0 |          -2 |       1 | 2024-12-02 07:31:46.587000 |
+------+-----------+--------+------+-------------+---------+----------------------------+
6 rows in set (0.03 sec)
```

## Checklist

- [x]  I have added the necessary unit tests and integration tests.

## Refer to a related PR or issue link (optional)
https://github.com/GreptimeTeam/greptimedb-ingester-go/issues/54

### CODE REVIEW

**Comment by zyy17** [comment]
IMO, the meaning of `greptime:"-"` should be: **ignore this field when writing and read with struct**.

For this PR, it just writes the zero value, it's not a field ignore.

**Comment by daviderli614** [comment]
> IMO, the meaning of `greptime:"-"` should be: **ignore this field when writing and read with struct**.
> 
> For this PR, it just writes the zero value, it's not a field ignore.

Wait for me to modify it soon.

**Comment by daviderli614** [comment]
Make some changes, please review again. @zyy17 
Usage Examples: [README](https://github.com/GreptimeTeam/greptimedb-ingester-go/blob/abb78211f1236d69d0664745eac4beb58e8e2fe2/examples/ignorefields/README.md)
There is also a unit test `TestParseSchemaWithIgnoreFields` that has passed.

File: schema/schema.go [hunk 1]
```diff
@@ -142,15 +144,19 @@ func (s *Schema) parseValues(input any) error {
 		return fmt.Errorf("unsupported type %T of %+v", input, input)
 	}
 
-	size := len(reflect.VisibleFields(typ))
-	values := make([]*gpb.Value, 0, size)
-	for i, structField := range reflect.VisibleFields(typ) {
-		if !structField.IsExported() {
+	visibleFields := reflect.VisibleFields(typ)
+	size := make([]reflect.StructField, 0, len(visibleFields))
```
Comment (zyy17) [inline schema/schema.go:148]: `size` is not a proper name. Maybe we can use `processingFields`

File: schema/schema.go [hunk 2]
```diff
@@ -142,15 +144,19 @@ func (s *Schema) parseValues(input any) error {
...
+	values := make([]*gpb.Value, 0, len(size))
+
+	for _, structField := range visibleFields {
+		if !structField.IsExported() || structField.Tag.Get("greptime") == "-" {
```
Comment (zyy17) [inline schema/schema.go:152]: `-` can be extracted as a const variable.

File: schema/field.go [hunk 3]
```diff
@@ -52,6 +52,10 @@ func newColumnSchema(columnName string, semanticType gpb.SemanticType, datatype
 func parseField(structField reflect.StructField) (*Field, error) {
 	tags := parseTag(structField)
 
+	if _, ok := tags["-"]; ok && len(tags) == 1 {
```
Comment (zyy17) [inline schema/field.go:55]: `-` can be extracted as a const variable; maybe we can use `IgnoreFiledTag`.

File: examples/skipfields/main.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
+// Copyright 2024 Greptime Team
```
Comment (zyy17) [inline examples/skipfields/main.go:1]: Do you think we should need an example for this feature? It's a very small feature and we can share one example for multiple features.

File: examples/skipfields/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
+# Insert data into GreptimeDB
+
+When fields are marked with `greptime:"-"`, writing to these fields will be skipped.
```
Comment (zyy17) [inline examples/skipfields/README.md:3]: It's better to use `ignored` instead of `skipped`.

File: schema/schema_test.go [hunk 6]
```diff
@@ -975,3 +975,288 @@ func TestParseWithUnmatchedDatatype(t *testing.T) {
 		assert.Nil(t, tbl)
 	}
 }
+
+func TestParseSchemaWithSkipFields(t *testing.T) {
```
Comment (zyy17) [inline schema/schema_test.go:979]: Is it necessary to write such a complicated test case? Can we make it small and clean? It will be easy to maintain.

**Review by zyy17 — APPROVED** [review]
LGTM

**Pull request merged**